app/code/Magento/Eav/Test/Unit/Model/TypeLocator/SimpleTypeTest.php

```diff
@@ -0,0 +1,139 @@
+use Magento\Eav\Model\AttributeRepository;
+use Magento\Framework\Webapi\CustomAttribute\ServiceTypeListInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Catalog\Model\ResourceModel\Eav\Attribute;
+use Magento\Framework\Reflection\TypeProcessor;
+
+/**
+ * Class SimpleTypeTest
+ *
+ * @package Magento\Eav\Test\Unit\Model\TypeLocator
+ */
+class SimpleTypeTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var ObjectManager
+     */
+    private $objectManger;
+
+    protected function setUp()
+    {
+        $this->objectManger = new ObjectManager($this);
+    }
+
+    /**
+     * Test testGetType method
+     *
+     * @param ServiceTypeListInterface|\PHPUnit_Framework_MockObject_MockObject $serviceTypeListMock
+     * @param string $getFrontendInputReturn
+     * @param string $getBackendTypeReturn
+     * @param string $expected
+     * @dataProvider getTypeDataProvider
+     */
+    public function testGetType($serviceTypeListMock, $getFrontendInputReturn, $getBackendTypeReturn, $expected)
+    {
+        /**
+         * @var Attribute|\PHPUnit_Framework_MockObject_MockObject
+         */
+        $attributeMock = $this->getMockBuilder(Attribute::class)
+                                    ->disableOriginalConstructor()
+                                    ->setMethods(['getFrontendInput', 'getBackendType'])
+                                    ->getMock();
+        $attributeMock->expects($this->any())
+                            ->method('getFrontendInput')
+                            ->willReturn($getFrontendInputReturn);
+        $attributeMock->expects($this->any())
+                            ->method('getBackendType')
+                            ->willReturn($getBackendTypeReturn);
+
+        /**
+         * @var AttributeRepository|\PHPUnit_Framework_MockObject_MockObject
+         */
+        $attributeRepositoryMock = $this->getMockBuilder(AttributeRepository::class)
+                                              ->disableOriginalConstructor()
+                                              ->setMethods(['get'])
+                                              ->getMock();
+        $attributeRepositoryMock->expects($this->any())
+                                      ->method('get')
+                                      ->willReturn($attributeMock);
+
+        /**
+         * @var SimpleType|\PHPUnit_Framework_MockObject_MockObject
+         */
+        $simpleType = ($this->objectManger)->getObject(
+            SimpleType::class,
+            [
+                'attributeRepository' => $attributeRepositoryMock,
+                'serviceTypeList' => $serviceTypeListMock
+            ]
+        );
+
+        $this->assertSame($expected, $simpleType->getType('testAttributeCode', 'testEntityType'));
+    }
+
+    /**
+     * @return array
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     */
+    public function getTypeDataProvider()
```

Comment:
Such test does not actually verify anything. It's just a method logic written in a different form.

Try splitting into proper test cases.